### ISSUE TITLE
Not able to find Environmental Telemetry logs

### ISSUE THREAD

**Issue by jasgroupit** [Affirmatech/MeshSense#25]
I have several nodes setup with one main node to view the data of my other nodes which i use in a farm. but i am not able to see telemetry app logs regarding environmental data

**Comment by Kealper** [comment]
I'm just guessing but they may be talking about the telemetry coming from the environmental and power metrics in the Telemetry module. Stuff like temperature, humidity, pressure, current draw, etc., from I2C sensors that can be added.

**Comment by jasgroupit** [comment]
Yes exactly that, temperature humidity pressure etc. Is there a way to display this ?

**Comment by lupusworax** [comment]
Same Request and issue.
As of now a node that functions perfectly with a BME280  shows the following:

{
  "num": 3306578313,
  "user": {
    "id": "!c5166189",
    "longName": "Lupus Worax 6189",
    "shortName": "🐺",
    "macaddr": "5NrFFmGJ",
    "hwModel": "HELTEC_MESH_NODE_T114",
    "isLicensed": false,
    "role": "CLIENT",
    "publicKey": "R3Lx2m/TVSMQDLASID=8363dDvaQdmJrg8+Ho="
  },
  "position": {
    "latitudeI": 477885538,
    "longitudeI": 137496299,
    "altitude": 210,
    "time": 1731941595,
    "locationSource": "LOC_UNSET",
    "altitudeSource": "ALT_UNSET",
    "timestamp": 0,
    "timestampMillisAdjust": 0,
    "PDOP": 0,
    "HDOP": 0,
    "VDOP": 0,
    "gpsAccuracy": 0,
    "groundSpeed": 0,
    "groundTrack": 0,
    "fixQuality": 0,
    "fixType": 0,
    "satsInView": 0,
    "sensorId": 0,
    "nextUpdate": 0,
    "seqNumber": 0,
    "precisionBits": 32
  },
  "snr": 6.5,
  "lastHeard": 1731941591.13,
  "deviceMetrics": {
    "batteryLevel": 78,
    "voltage": 3.9730000495910645,
    "channelUtilization": 4.791666507720947,
    "airUtilTx": 0.2473333328962326,
    "uptimeSeconds": 98516
  },
  "channel": 0,
  "viaMqtt": false,
  "hopsAway": 0,
  "isFavorite": false,
  "rssi": -53,
  "trace": null
}

**Comment by Soltares** [comment]
Environmental data appears to come from the `environmentMetrics` Telemetry Message type.  Will attempt to capture this and attach to the tracked Node information.

**Comment by Soltares** [comment]
Successfully captured some data with the latest dev build!

```json
  "environmentMetrics": {
    "temperature": 15.008466720581055,
    "relativeHumidity": 53.141929626464844,
    "barometricPressure": 1004.330810546875,
    "gasResistance": 624.9585571289062,
    "iaq": 133
  }
  ```

**Comment by lupusworax** [comment]
> Successfully captured some data with the latest dev build!
> 
> ```json
>   "environmentMetrics": {
>     "temperature": 15.008466720581055,
>     "relativeHumidity": 53.141929626464844,
>     "barometricPressure": 1004.330810546875,
>     "gasResistance": 624.9585571289062,
>     "iaq": 133
>   }
> ```

Thats freaking awesome! 💯   cant wait to see it pop up with the next ( hopefully)  update

**Comment by Soltares** [comment]
Initial draft at displaying the data
![image](https://github.com/user-attachments/assets/290f3a42-ca33-44b5-84a9-14588e632d25)
Temp / Pressure / Gas Resistance / Humidity / Air Quality

**Comment by lupusworax** [comment]
> Initial draft at displaying the data ![image](https://private-user-images.githubusercontent.com/40914111/389083645-290f3a42-ca33-44b5-84a9-14588e632d25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzMDY3MTQsIm5iZiI6MTczMjMwNjQxNCwicGF0aCI6Ii80MDkxNDExMS8zODkwODM2NDUtMjkwZjNhNDItY2EzMy00NGI1LTg0YTktMTQ1ODhlNjMyZDI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDIwMTMzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhOGJjM2Q5Y2M4ZTQxNjFlN2IzMWMzZmMyZDI2ZjVhMGJhNjAwZDBkNDA5Y2VkYzg3Y2RlMDI3MTJkMWUyNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NascE1im3-IeVAHCEtzNKmoAAazqE2zExHH3eIqOtzo) Temp / Pressure / Gas Resistance / Humidity / Air Quality

You nailed it!

**Comment by Soltares** [comment]
This has been published in `v1.0.16-beta.2`!  Hopefully everything works well!

**Comment by lupusworax** [comment]
updated but so far no telemetry data from my solar node, could it take a while until it updates telemetry?

**Comment by Soltares** [comment]
Yes it's possible.  There was only one node in my local mesh that published data and it was on the edge of my network.  It is likely based on the environmentUpdateInterval setting of your solar node.

Device configuration of the MeshSense connected node can now be viewed in the Settings modal:

![image](https://github.com/user-attachments/assets/d52ce259-396d-4bc7-888d-49f2acd0eb51)

**Comment by lupusworax** [comment]
I see, yeah the connected node sadly does not have an environmental sensor. Well I give it a bit and will report back, as of now it looks like this: 
![image](https://github.com/user-attachments/assets/2e25b340-3e85-463b-879b-2a85e8f3e437)

**Comment by lupusworax** [comment]
Fuck yeah it works! 
![image](https://github.com/user-attachments/assets/ce23a440-4c14-4c2f-8511-7ed437912ee9)
amazing work! Thanks for taking the time to work  and adding this!

**Comment by Soltares** [comment]
Fantastic!!  Thanks for the report!  Looks like I need to potentially hide or clean up fields if they're missing.  Appreciate it!

**Comment by Soltares** [comment]
Closing as feature is released in `v1.0.16`
